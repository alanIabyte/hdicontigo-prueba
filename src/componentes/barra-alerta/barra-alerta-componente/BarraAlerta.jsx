import React from "react";
import PropTypes from "prop-types";
import { v4 } from "uuid";
import CloseRoundedIcon from "@material-ui/icons/CloseRounded";
import ErrorRoundedIcon from "@material-ui/icons/ErrorRounded";
import InfoRoundedIcon from "@material-ui/icons/InfoRounded";
import WarningRoundedIcon from "@material-ui/icons/WarningRounded";
import CheckCircleRoundedIcon from "@material-ui/icons/CheckCircleRounded";
import {
  EnvolvedorBarraAlerta,
  BarraAlertaContenedor,
  BarraAlertaIcono,
  BarraAlertaTexto,
  BarraAlertaCierre,
} from "./BarraAlerta.styled";

const BarraAlerta = (props) => {
  const {
    clase,
    etiqueta,
    estilo,
    mostrarAlerta,
    manejarCierre,
    sinEncabezado,
    fijo,
    posicionAbsoluta,
    encabezadoAlto,
  } = props;
  let icono;
  switch (estilo) {
    case "error":
      icono = <ErrorRoundedIcon />;
      break;
    case "notificacion":
      icono = <InfoRoundedIcon />;
      break;
    case "exitoso":
      icono = <CheckCircleRoundedIcon />;
      break;
    default:
      icono = <WarningRoundedIcon />;
      break;
  }

  const despliegaTextoCuerpo = () => ({ __html: etiqueta });

  return (
    <EnvolvedorBarraAlerta
      key={v4()}
      mostrar={mostrarAlerta}
      estilo={estilo}
      className={clase}
      sinEncabezado={sinEncabezado}
      fijo={fijo}
      posicionAbsoluta={posicionAbsoluta}
      encabezadoAlto={encabezadoAlto}
    >
      <BarraAlertaContenedor>
        <BarraAlertaIcono>{icono}</BarraAlertaIcono>
        <BarraAlertaTexto dangerouslySetInnerHTML={despliegaTextoCuerpo()} />
        <BarraAlertaCierre
          type="submit"
          onClick={() => {
            manejarCierre(false);
          }}
        >
          <CloseRoundedIcon />
        </BarraAlertaCierre>
      </BarraAlertaContenedor>
    </EnvolvedorBarraAlerta>
  );
};

BarraAlerta.propTypes = {
  clase: PropTypes.string,
  estilo: PropTypes.oneOf(["alerta", "error", "notificacion", "exitoso"]),
  etiqueta: PropTypes.string,
  manejarCierre: PropTypes.func,
  mostrarAlerta: PropTypes.bool,
  sinEncabezado: PropTypes.bool,
  fijo: PropTypes.bool,
  posicionAbsoluta: PropTypes.bool,
  encabezadoAlto: PropTypes.bool,
};

BarraAlerta.defaultProps = {
  clase: "",
  estilo: "error",
  etiqueta: "",
  manejarCierre() {},
  mostrarAlerta: false,
  sinEncabezado: false,
  fijo: false,
  posicionAbsoluta: false,
  encabezadoAlto: false,
};

export default BarraAlerta;
