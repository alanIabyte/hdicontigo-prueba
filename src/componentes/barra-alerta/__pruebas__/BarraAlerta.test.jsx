import React from "react";
import renderer from "react-test-renderer";
import BarraAlerta from "../barra-alerta-componente/BarraAlerta";

describe("#BarraAlerta", () => {
  it("se despliega correctamente", () => {
    const tree = renderer.create(<BarraAlerta />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
