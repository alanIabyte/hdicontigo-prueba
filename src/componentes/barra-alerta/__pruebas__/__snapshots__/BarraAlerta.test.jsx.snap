// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#BarraAlerta se despliega correctamente 1`] = `
<div
  className="sc-aXZVg cEfDuC"
>
  <div
    className="sc-gEvEer gyAAKl"
  >
    <div
      className="sc-eqUAAy eiYtLT"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 11c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v4c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
        />
      </svg>
    </div>
    <span
      className="sc-fqkvVR kWIBUX"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
    <button
      className="sc-dcJsrY hPllxL"
      onClick={[Function]}
      type="submit"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18.3 5.71a.9959.9959 0 00-1.41 0L12 10.59 7.11 5.7a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
        />
      </svg>
    </button>
  </div>
</div>
`;
